deepmd/dpmodel/descriptor/dpa4.py

```
@@ -0,0 +1,1258 @@
+        seed_out = child_seed(self.seed, 2)
+        seed_radial_embedding = child_seed(self.seed, 3)
+        seed_env_seed = child_seed(self.seed, 4)
+
+        # === L/M schedules ===
+        self._init_lm_schedules(lmax, n_blocks, l_schedule, mmax, m_schedule)
+        self.kmax = int(kmax)
+        if self.kmax < 0:
+            raise ValueError("`kmax` must be non-negative")
+        if self.kmax > self.lmax:
+            raise ValueError("`kmax` must be <= `lmax`")
+        self.ebed_dims = [get_so3_dim_of_lmax(l) for l in self.l_schedule]
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)